### PR TITLE
Remove member when shutting down LifecycleEvent is triggered.

### PR DESCRIPTION
**Pull request by vmaraloiu**

### CODE REVIEW

**Comment by dpwspoon** [comment]
Logic wise this looks good to me.  Please add at least one test, though it would be great if you can come up with a couple different scenarios.

File: server/src/test/java/org/kaazing/gateway/server/context/resolve/DefaultClusterContextTest.java [hunk 1]
```diff
@@ -281,8 +282,8 @@ public void shouldSeeBalancerStateAfterClusterMembersLeave() throws Exception {
                     connects4,
                     null);
 
-            startClusterContext(clusterContext1);
-            addToClusterState(clusterContext1.getCollectionsFactory(), clusterContext1.getLocalMember(), 1);
+            Map<ClusterContext, String> clusterContextToServiceUrlMap = new HashMap<>();
```
Comment (Anisotrop) [inline server/src/test/java/org/kaazing/gateway/server/context/resolve/DefaultClusterContextTest.java]: This holds the expected balanced services for each node. Should be renamed accordingly.

File: server/src/test/java/org/kaazing/gateway/server/context/resolve/DefaultClusterContextTest.java [hunk 2]
```diff
@@ -407,58 +409,33 @@ private void addToClusterState(CollectionsFactory factory, MemberId memberId, in
...
-                fail("Expected 4 URIs as balance targets in cluster member 1's memory");
-            } else {
-                List<String> myBalanceTargets = new ArrayList<>();
-                myBalanceTargets.add("ws://node1.example.com:8080/path");
-                if (!validateMemberIdBalancerMap(clusterContext[0].getCollectionsFactory(), clusterContext[0]
-                        .getLocalMember(), myBalanceTargets)) {
-                    fail("Expected ws://node1.example.com:8080/path as balance target provided by cluster member 1");
-                }
-            }
+    private void validateClusterState(List<ClusterContext> activeMembers, Map<ClusterContext, String> clusterContextToServiceUrlMap) {
```
Comment (Anisotrop) [inline server/src/test/java/org/kaazing/gateway/server/context/resolve/DefaultClusterContextTest.java]: activeMembers should be renamed to expectedActiveMembers and clusterContextToServiceUrlMap should be renamed to expectedClusterContextToServiceUrlMap

File: server/src/test/java/org/kaazing/gateway/server/context/resolve/DefaultClusterContextTest.java [hunk 3]
```diff
@@ -485,6 +462,14 @@ private boolean validateMemberIdBalancerMap(CollectionsFactory factory, MemberId
         String balanceURI = "ws://www.example.com:8080/path";
         List<String> memberBalanceTargets = balancerMap.get(balanceURI);
         return (memberBalanceTargets != null) && memberBalanceTargets.containsAll(balanceTargets);
+    }
+
+    private boolean validateMemberIdBalancerMapKeyset(CollectionsFactory factory, MemberId memberId, Set<MemberId> activeMembers) {
```
Comment (Anisotrop) [inline server/src/test/java/org/kaazing/gateway/server/context/resolve/DefaultClusterContextTest.java]: Parameter MemberId memberId is not used

File: server/src/test/java/org/kaazing/gateway/server/context/resolve/DefaultClusterContextTest.java [hunk 4]
```diff
@@ -467,7 +444,7 @@ private boolean validateSharedBalancer(CollectionsFactory factory, Set<String> b
         Map<String, Set<String>> sharedBalancerMap = factory.getMap(BALANCER_MAP_NAME);
```
Comment (Anisotrop) [inline server/src/test/java/org/kaazing/gateway/server/context/resolve/DefaultClusterContextTest.java]: Should use constant defined in org.kaazing.gateway.server.context.resolve.DefaultServiceContext

**Comment by vmaraloiu** [comment]
@Anisotrop Thanks for the feedback.

**Pull request merged**